libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/LoginActivity.kt

```diff
@@ -164,14 +169,19 @@ open class LoginActivity : AppCompatActivity(), OAuthWebviewHelperEvents {
 
         salesforceSDKManager.setViewNavigationVisibility(this)
 
-        // Get login options from the intent's extras
-        val loginOptions = fromBundleWithSafeLoginUrl(intent.extras)
+        // Determine login options for QR code login or the app's usual login.
+        val loginOptions = when {
+            isDeepLinkedQrCodeLogin -> salesforceSDKManager.loginOptions
+            else -> fromBundleWithSafeLoginUrl(intent.extras)
+        }
 
         // Protect against screenshots
         window.setFlags(FLAG_SECURE, FLAG_SECURE)
 
-        // Fetch authentication configuration if required
-        salesforceSDKManager.fetchAuthenticationConfiguration()
+        // Fetch authentication configuration except for QR code login.
+        if (!isDeepLinkedQrCodeLogin) {
+            salesforceSDKManager.fetchAuthenticationConfiguration()
+        }
```

Comment:
@wmathurin - Do you recall the logic on this one?

Reply:
I took it out during the spike because it uses the selected login server which might not match the one specified in the QR url. To keep it, we have to refactor fetchAuthenticationConfiguration to optionally accept a loginServer and during a QR login, we should pass the loginServer across.

Reply:
With this as it is, are we skipping a task somehow during a deep linked QR Code Log In?  Also, I usually thought of a "deep linked" QR code log in as coming from an external QR code reader but the change in server would apply to internally read QR codes as well, I reckon.

Reply:
I was trying to make sure I understood this one.  With the code as we have it, one can only log in via a QR Code/Front Door Bridge from the connected app configured in `bootconfig.xml`.  @wmathurin, isn't that a constraint and if so we wouldn't need this check.

Otherwise, do have a need to support QR codes that would log the user into something they couldn't log into via the web view? Sorry, I'm just making sure I know the scope.

Reply:
In the spike, I was extracting the login server from the QR url - see https://github.com/forcedotcom/SalesforceMobileSDK-Android/commit/7468f20a6a052767da718ec2c1a1c05f8422c332.

Reply:
@wmathurin - I can see the current code you and I have here, since this includes your spike commits, does the URL extraction from the QR code and sets login URL on login options.  The trouble is that fails to log in.  Say, if I set the app's default login URL to "Production"/`https://login.salesforce.com` and then use a QR code from `https://mobilesdkatsdb6--c.test1.vf.pc-rnd.force.com/apex/qrcodelogin`.  The web view displays "Webpage Not Available"/"The webpage at mobilesdk://android/pn/tester?code=... could not be loaded because: net:ERR_UNKNOWN_URL_SCHEME"

If I recall the code, we'd need to also have the connected app's callback URL so the auth flow would be able to handle the authorization response page.  Do we want to do that and is this actually a realistic scenario?

Just for fun, I did do a little switch users magic and it _is_ possible to sign in to the app across two different login URLs, but I wasn't sure we actually support that.

Reply:
Also, as the code is now this boolean is set from `isDeepLinkedQrCodeLogin`.  That method probably needs to change in the follow-up pull request for [W-16336722](https://gus.lightning.force.com/a07EE00001xTbt7YAC).  How the QR Code payload is structured is theoretically app to the APEX/VisualForce Page set up by the MSDK user, so that method should probably move to the app/template code since so the app can interpret the deep link intent and pass explicit options to MSDK.

Reply:
An app can disallow users from adding login servers by hiding the button to pick servers.
However when the app does not hide the button, users are allowed to add login servers.

When allowed, if QR points to a login server they don't have or haven't selected, I don't think it would be a good user experience to require the user to first add the login server then scan the QR code. It kind of defeats the point of having QR as an easier way to log in.

When the button is hidden, if we extract the login server from the QR URL like I did in the spike, it would allow the user to change login server even though the app thought it was prohibiting it.

The correct behavior should probably be:
When doing a login from QR URL, extract the login server, check if the login server is part of the list of login servers or if adding login servers is allowed. If that is true, proceed with login (including the fetching of the auth config).
If the login server is not part of the list of login servers and adding login servers is disallowed, fail with a message to the server indicating the problem ("QR uses a login server that is not configured in mobile application").

If that's too much for 12.2, we could create a new story for it and simply document in 12.2 what's not supported.

Reply:
I broke this out into [W-16893221](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000022ReOAYA0/view).